### DISCUSSION TITLE
Setting image heights in `layout-ncol` for PDF output

### DISCUSSION THREAD

**Discussion by ollyhensby** [quarto-dev/quarto-cli#11714] [Q&A] [answered]
### Description

The image heights are not respected when used within a `layout-ncol`. Am I using `layout-ncol` incorrectly? Am I supposed to set the height of the `layout-ncol` instead of the images themselves?

Example:

```qmd
---
title: Image Height Issue in Column Layout
author: Oliver Hensby
format:
    pdf:
        pdf-engine: xelatex
---

## Some Images

![](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){height=30mm}

![](https://www.gstatic.com/webp/gallery3/2_webp_a.png){height=10mm}

::: {layout-ncol=2}
![](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){height=30mm}

![](https://www.gstatic.com/webp/gallery3/2_webp_a.png){height=10mm}
:::
```

Quarto version: 1.6.37
OS: Ubuntu 22.04.3 LTS

**Comment by mcanouil** [comment]
Have you read the documentation on figure sizing?

https://quarto.org/docs/authoring/figures.html#figure-sizing

I don't believe your example works in any scenario.
The correct arguments are `height` and `width`.

**Reply by ollyhensby** [reply]
I did read the documentation on figure sizing but maybe I am missing something.

I can set `height` like this for example:

```qmd
---
title: Test
format:
    pdf:
        pdf-engine: xelatex
---

![](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){height=30mm}
```

and the height is applied but when I place this within a `layout-ncol` block, the height is not applied anymore:

```qmd
---
title: Test
format:
    pdf:
        pdf-engine: xelatex
---

::: {layout-ncol=1}
![](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){height=30mm}
:::
```

I noticed that if I add a caption to the figure the height defined is respected again:

```md
---
title: Test
format:
    pdf:
        pdf-engine: xelatex
---

::: {layout-ncol=1}
![Test](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){height=30mm}
:::
```

I finally tried with the `width` attribute and this is applied in the case where there is no caption in the `layout-ncol` block:

```md
---
title: Test
format:
    pdf:
        pdf-engine: xelatex
---

::: {layout-ncol=1}
![](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){width=30mm}
:::
```

It seems only the `width` attribute is respected in the `layout-ncol` block but not the `height` attribute. And also, when just using `height`, it is only applied when there is a caption in the `layout-ncol`.

**Reply by mcanouil** [reply]
It seems to work just fine so there is a slight difference which occurs in LaTeX somewhere.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: pdf
---

This is a playground for Quarto.

::: {layout-ncol=2}
![An image]({{< placeholder 60 40 >}}){height=20mm}

![An image]({{< placeholder 60 40 >}}){height=40mm}
:::

![An image]({{< placeholder 60 40 >}}){#fig-placeholder-3 height=20mm}

![An image]({{< placeholder 60 40 >}}){#fig-placeholder-4 height=40mm}

````

</td><td>

<img width="423" alt="image" src="https://github.com/user-attachments/assets/84a567a2-8801-4b22-89c2-9c61a08e83ba" />

</td></tr>
</table>

**Reply by mcanouil** [reply]
The only difference I see is the caption.

```qmd
---
title: "Quarto Playground"
format: pdf
keep-tex: true
---

::: {layout-ncol="1"}
![](https://quarto.org/quarto.png){height=30mm}
:::

::: {layout-ncol="1"}
![Quarto](https://quarto.org/quarto.png){height=30mm}
:::
```

The resulting LaTeX file (`keep-tex: true`):


```latex
\begin{figure}

\begin{minipage}{\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{index_files/mediabag/quarto.png}}\end{minipage}%

\end{figure}%

\begin{figure}

\begin{minipage}{\linewidth}

\begin{figure}[H]

{\centering \includegraphics[width=\linewidth,height=30mm,keepaspectratio]{index_files/mediabag/quarto.png}

}

\subcaption{Quarto}

\end{figure}%

\end{minipage}%

\end{figure}%
```

I believe that's by design as the behaviour is the same across formats (HTML, Typst, LaTeX), but since no caption is required outside of layout div it might be a bug.
CC @cscheid

**Reply by cscheid** [reply, marked as answer]
This is a bug that @hadley pointed out to me just earlier today. We'll fix.

**Reply by mcanouil** [reply]
Thanks for confirming.

**Reply by ollyhensby** [reply]
Thank you both for looking into this!

**Reply by ollyhensby** [reply]
@cscheid Sorry to be a bit cheeky but do you have an idea of when this bug will be addressed? I am also happy to have a go resolving the issue if you point me to the whereabouts of the bug. Thank you.

**Comment by ollyhensby** [comment]
For anyone that stumbles onto this discussion, I have a work around for this for now.

I add `&nbsp;` (non-breaking space) as the captions so the heights are applied.
Then in the header metadata I have included the latex code to remove the label format from figures and subfigures.

See an example below of the work around:

```md
---
title: Image Height Issue in Column Layout
author: Oliver Hensby
format:
    pdf:
        pdf-engine: xelatex
        include-in-header:
            - text: |
                \usepackage{caption}
                \usepackage{subcaption}
                \captionsetup[figure]{labelformat=empty}
                \captionsetup[subfigure]{labelformat=empty}
---

## Some Images

::: {layout-ncol=2}
![&nbsp;](https://www.gstatic.com/webp/gallery3/1_webp_ll.png){height=30mm}

![&nbsp;](https://www.gstatic.com/webp/gallery3/2_webp_a.png){height=10mm}
:::
```

Output:

![image](https://github.com/user-attachments/assets/bf8f5ac5-89a2-4558-aa5c-a782f1745b5a)
